src/cart/crt_hg.c

```diff
@@ -664,11 +714,13 @@ crt_hg_class_init(int provider, int idx, hg_class_t **ret_hg_class)
 		D_GOTO(out, rc = -DER_HG);
 	}
 
-	rc = crt_hg_get_addr(hg_class, addr_str, &str_size);
-	if (rc != 0) {
-		D_ERROR("crt_hg_get_addr() failed, rc: %d.\n", rc);
-		HG_Finalize(hg_class);
-		D_GOTO(out, rc = -DER_HG);
+	if (crt_is_service() || !crt_na_type_is_ucx(provider)) {
+		rc = crt_hg_get_addr(hg_class, addr_str, &str_size);
```

Comment:
same question here as above, what about the UCX case? addr_str is supposed to be stored in the context, and later used in various places